internal/kallsyms/kallsyms.go

```diff
@@ -56,7 +56,12 @@ func loadKernelModuleMapping(f io.Reader) (map[string]string, error) {
 	mods := make(map[string]string)
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
-		fields := bytes.Fields(scanner.Bytes())
+		line := scanner.Bytes()
+		// bytes.Fields is expensive, so filter for '[' first
+		if !bytes.ContainsRune(line, '[') {
```

Comment:
I am not that sure (since I don't know the format that well), but if the closing bracket `]` is always at the end of the line we could utilize that to improve performance even further

Reply:
Do you mean replacing `bytes.ContainsRune` by something like this?
```go
if strings.HasSuffix(line, "]")
```

That should filter the same amount of lines as `bytes.ContainsRune`, so it's a matter of if `strings.HasSuffix` is faster than `bytes.ContainsRune`. I have not tested it but I suspect it is similar for short lines like the ones in kallsyms, and in any cases would not change the amount of memory allocations.

The printfs for /proc/kallsyms are implemented here:
https://github.com/torvalds/linux/blob/eca631b8fe808748d7585059c4307005ca5c5820/kernel/kallsyms.c#L446-L465

The last change was 3 years ago to add the buildid inside the brackets:
https://github.com/torvalds/linux/commit/9294523e3768030ae8afb84110bcecc66425a647#diff-9538b26d3e082f233e6adac664cd2c14cbf2d510d5d7f286eef329c58de87eadR451
This is only for kernels with `CONFIG_STACKTRACE_BUILD_ID` (which is not set for my kernel). But the way cilium/ebpf parses the line is fine with regards to that change.

So I would say the format seems more or less stable but can change. And testing "contains-[" should be more stable than testing "suffix-]", in case Linux adds new fields at the end.